wheatley/row_generation/complib_composition_generator.py

```diff
@@ -58,27 +59,38 @@ def removeprefix(string: str, prefix: str) -> str:
 class ComplibCompositionGenerator(RowGenerator):
     """ The RowGenerator subclass for generating rows from a CompLib composition. """
 
-    complib_url = "https://complib.org/composition/"
+    complib_url = "https://api.complib.org/composition/"
 
     def __init__(self, comp_id: int, access_key: Optional[str]=None) -> None:
+        # Generate URL and request the rows
         url = self.complib_url + str(comp_id) + "/rows"
         if access_key:
             url += "?accessKey=" + access_key
         request_rows = requests.get(url)
-
+        # Check for the status of the requests
         if request_rows.status_code == 404:
             raise InvalidCompError(comp_id)
-
         if request_rows.status_code == 403:
             raise PrivateCompError(comp_id)
-
         request_rows.raise_for_status()
+        # Parse the request responses as JSON
+        response_rows = json.loads(request_rows.text)
 
         # New line separated, skip the first line (rounds)
-        split_rows = request_rows.text.splitlines(False)[1::]
-        self.loaded_rows: List[Row] = [Row([Bell.from_str(bell) for bell in row]) for row in split_rows]
+        split_rows = response_rows['rows']
```

Comment:
Looking at 
https://complib.org/composition/49603/rows vs https://api.complib.org/composition/49603/rows

I think this now needs to be skip the first 2 instead of skip none.
Though both end in rounds.

Reply:
Oh yes good point - I forgot about that.

Stripping the first two rows also handles backstroke starts correctly (if not very cleanly) because CompLib's API returns them as starting with 3 rows of rounds (see [this comp](https://api.complib.org/composition/68084/rows)).  But it would be nice to be able to do other things like specify the start row to `MethodRowGenerator` which might cause a backstroke start in other row generators.  Therefore, the stroke of the start of a composition should, I think, be made part of the interface of `RowGenerator`.  I'll probably handle that in #119 since we're overhauling the row gen anyway.

For the time being, this implementation will suffice.